src/index.js

```diff
@@ -6,6 +6,22 @@ let L = 'en' // global locale
 const Ls = {} // global loaded locale
 Ls[L] = en
 
+/** Index to get the number of days */
+const MONTH_DAYS = {
+  0: 31, // January
+  1: 28, // February
+  2: 31, // March
+  3: 30, // April
+  4: 31, // May
+  5: 30, // June
+  6: 31, // July
+  7: 31, // August
+  8: 30, // September
+  9: 31, // October
+  10: 30, // November
+  11: 31 // December
+}
+
```

Comment:
Introducing the MONTH_DAYS constant might increase the bundle size, which may not be an ideal optimization?

Reply:
I've noticed this PR introduces some additional code into both `utils` and `index`.

A core principle of Day.js is its minimal bundle size, and we're highly sensitive to any changes that might increase it. For a change like this to be a worthwhile trade-off, we would need to see a significant performance improvement over the current `endOf` implementation.

Have you conducted any performance benchmarks to demonstrate the gains of this new approach? This data would be very helpful for us to evaluate if the change is a net positive for the project.

Looking forward to your thoughts!